### ISSUE TITLE
Ability to use RHF with dynamic forms.

### ISSUE THREAD

**Issue by kapsule-studio** [react-hook-form/react-hook-form#10808]
What I'm wishing for is a solution to solve the problem I exposed here: https://github.com/orgs/react-hook-form/discussions/10793

TLDR: I have multiple forms that are duplicates. Only one form is displayed at a time. How can I call for `useForm()` or `useFormContext()` in each Form component and renaming the properties from it so they don't just override themselves. 

The setup I have now is working fine, I managed to build a workaround to separate the errors for each Form. The issue is that I don't know before hand how many forms will be there. Users can add and remove them. So the solution of renaming each `register` from `useForm()` to `register1`, `register2` and so on is not an option. 

All the examples I saw during my research where involving multi forms that you already know the count of. I don't. And since it is impossible to rename a distructured property to a dynamic value, I'm just stuck...

For examples, each of my Forms has a unique id that it receives as props. I can't do something like this to rename each `register`:

``` tsx
const {register: `register-${id}`} = useForm()
```

Or even something like this : 

``` tsx
const uniqueRegister = `register-${id}`

const {register: uniqueRegister} = useForm()
// I can't do that because it would be declaring `uniqueRegister` twice...
```

So what should be the right approach there? Any help would be greatly appreciated, even if it's to tell me that this is simply impossible using RHF. At least I'll know I have to find another way/library.
